src/rest/APIRequest.js

```diff
@@ -46,11 +47,20 @@ class APIRequest {
       headers['Content-Type'] = 'application/json';
     }
 
+    const controller = new AbortController();
+    const timeout = setTimeout(() => controller.abort(), 10000);
     return fetch(url, {
       method: this.method,
       headers,
       agent,
       body,
+      signal: controller.signal,
+    }).then(res => {
```

Comment:
I suppose this could use [`Promise#finally`][Promise.prototype.finally], since it seems to be available in node 10, see [node.green].

[Promise.prototype.finally]:https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/finally
[node.green]:https://node.green/#ES2018-features-Promise-prototype-finally